hazelcast/src/main/java/com/hazelcast/spi/impl/proxyservice/impl/ProxyServiceImpl.java

```diff
@@ -248,7 +248,7 @@ public void dispatchEvent(final DistributedObjectEventPacket eventPacket, Object
         } else {
             final ProxyRegistry registry = registries.get(serviceName);
             if (registry != null) {
-                registry.destroyProxy(eventPacket.getName(), false);
+                registry.destroyProxy(eventPacket.getName(), true);
```

Comment:
`true` means publish events after destroying the proxy. Is this intentional?

Reply:
good catch!